doc/unions.asciidoc

```diff
@@ -0,0 +1,155 @@
+== Unions
+
+Golo allows the definition of sum algebraic data types, also known as
+_tagged_union_
+footnote:[ http://en.wikipedia.org/wiki/Tagged_union[Wikipedia: Tagged union]
```

Comment:
I have used footnotes everywhere in the doc I wrote (augmentations, decorators)... I'll review that and change to simple links